src/d3a/models/strategy/load_hours.py

```diff
@@ -111,15 +116,25 @@ def _init_price_update(self, fit_to_limit, energy_rate_increase_per_update, upda
                                  update_interval=update_interval, rate_limit_object=min)
         self.fit_to_limit = fit_to_limit
 
-    def _validate_rates(self):
-        for time_slot in generate_market_slot_list():
-            rate_change = None if self.fit_to_limit else \
-                self.bid_update.energy_rate_change_per_update[time_slot]
-            validate_load_device_price(
-                initial_buying_rate=self.bid_update.initial_rate[time_slot],
-                energy_rate_increase_per_update=rate_change,
-                final_buying_rate=self.bid_update.final_rate[time_slot],
-                fit_to_limit=self.bid_update.fit_to_limit)
+    @staticmethod
+    def _validate_rates(initial_rate, final_rate, energy_rate_change_per_update,
+                        fit_to_limit, init=False):
+        try:
+            for time_slot in generate_market_slot_list():
```

Comment:
I would very much prefer to use `enumerate`, or even validate each parameter separately, instead of generating the market slot list and try to access each element. The disadvantage of this method is that it is obviously not applicable for Canary Network, since the market slot list is going to be infinite. Furthermore, this way entries in the initial/final rate that are not market slots are not evaluated (not useful for now, but will be if we decide to change the slot_length via a live event).

Reply:
I am not sure what you mean here. I understand that this approach is not applicable for the CN, however, the user will probably also provide profiles in the CN case. And yes, `generate_market_slot_list` will not be usable here. I think this needs some more discussion actually, but only if we decide to use template strategies in the CN. Or didn't i get it right?